src/nvim/fold.c

```diff
@@ -457,6 +457,27 @@ void newFoldLevel(void)
   }
 }
 
+void foldChanged(win_T *wp)
+{
+  if (wp->w_lines_valid != 0) {
+    // Only trigger once per window redraw
+    char winid[NUMBUFLEN];
+    vim_snprintf(winid, sizeof(winid), "%d", wp->handle);
+    apply_autocmds(EVENT_FOLDCHANGED, winid, winid, false, wp->w_buffer);
+  }
+
+  // redraw
+  changed_window_setting_win(wp);
+}
+
+void foldEnable(win_T *wp, bool value)
+{
+  if (wp->w_p_fen != value) {
+    wp->w_p_fen = value;
+    foldChanged(wp);
+  }
+}
```

Comment:
Please read the control flow of `nv_zet()` more carefully. This function is not needed at all. Because the block at the end of `nv_zet()` is enough. What is actually needed is triggering `FoldChanged` in option callbacks.